### DISCUSSION TITLE
Raycasting through convex objects

### DISCUSSION THREAD

**Discussion by Pindrought** [jrouwe/JoltPhysics#1204] [Q&A] [answered]
I am currently trying to raycast to and through an object.
![0wT9J9g](https://github.com/user-attachments/assets/17045579-d9a3-4392-b407-00e860b2fbd8)

This works great for getting the ray to the object's surface, but once I try to cast a ray that continues from the inside of the shape, I am just getting a hit at the same position as the ray's origin.

I saw in the comments for CastRay
**"Convex objects will be treated as solid (meaning if the ray starts inside, you'll get a hit fraction of 0)"**

I am wanting to find a way to be able to cast rays through convex shapes from the inside and get the point of the collision through to the next flat surface of the shape.
Ex. The green ray would be from a 2nd raycast, other red line from a 3rd raycast, etc.
![image](https://github.com/user-attachments/assets/c10e5d96-0a5d-4c91-a6de-cc382bd1e227)

Is it possible to treat convex shapes as hollow during a raycast or how should I be handling this?

**Comment by jrouwe** [comment, marked as answer]
Use the 2nd CastRay function and set `RayCastSettings::mTreatConvexAsSolid = false`.

**Reply by Pindrought** [reply]
Worked perfect! Thanks.
![image](https://github.com/user-attachments/assets/ff45fd24-23ab-48c4-8e3d-701b9af29f35)
